analyzers/src/SonarAnalyzer.CSharp/Rules/ExceptionsShouldBeLogged.cs

```diff
@@ -112,8 +149,15 @@ public override void VisitInvocationExpression(InvocationExpressionSyntax node)
         }
 
         private static bool IsLoggingInvocation(InvocationExpressionSyntax invocationSyntax, SemanticModel model) =>
```

Comment:
> if (IsExceptionLogged) return;

Zsolt suggested the same thing and I've updated this in the base PR

Reply:
> Also, can you explain to me like#138, what is the idea with hasWhenFilter check?
> Is it a best-effort check, since we cannot parse the expression of the when filter? (unless we do SE, I guess)

Yes, this is to avoid raising FPs. I've improved the precision a bit. The main problem occurs when there are variable declarations when the filtering is done. In that case any exception logged is good enough.